pydantic_settings/sources.py

```diff
@@ -1996,19 +1997,20 @@ def _metavar_format_recurse(self, obj: Any) -> str:
             return '...'
         elif isinstance(obj, Representation):
             return repr(obj)
-        elif isinstance(obj, typing_extensions.TypeAliasType):
+        elif typing_objects.is_typealiastype(obj):
             return str(obj)
 
-        if not isinstance(obj, (typing_base, WithArgsTypes, type)):
+        origin = get_origin(obj)
```

Comment:
This function is still a big mess, in the future I'll add a string repr function in `typing-inspection`